fiwareskuld/users_management.py

```diff
@@ -0,0 +1,456 @@
+        if user:
+            self.update_domain_to_role(user, role_name, date_now)
+            self.update_quota(user, role_name)
+        return user
+
+    def update_user(self, user, role_name, date=None):
+        """
+        It updates the users.
+        :param user: the user
+        :param date: the date
+        :return: nothing
+        """
+        self.update_domain_to_role(user, role_name, date)
+        self.update_quota(user, role_name)
+
+    def delete_user(self, user):
+        """
+        It deletes the user
+        :param user: the user to be deleted.
+        :return: nothing
+        """
+        users = self.keystone.users.list(username=user.username)
+        if users or len(users) > 0:
+            # We delete the projects belonging to the user
+            projects = self.keystone.projects.list(user=users[0])
+            for project in projects:
+                self.keystone.projects.delete(project)
+            self.keystone.users.delete(users[0])
+        else:
+            raise Exception(404, "User {0} not found".format(user.username))
+
+    def delete_user_id(self, user_id):
+        """
+        It delete the user by its user_id
+        :param user_id: the user_id
+        :return: nothing
+        """
+        users = self.keystone.users.list(username=user_id)
+        if users or len(users) > 0:
+            self.delete_user(users[0])
+        else:
+            raise Exception(404, "User {0} not found".format(user_id))
+
+    def delete_trial_users(self):
+        """
+        It deletes the trial users.
+        :return:
+        """
+        users_trial = self.exp.get_trial_users()
+        for user in users_trial:
+            self.delete_user(user)
+
+    def delete_community_users(self):
+        """
+        It deletes the community users.
+        :return:
+        """
+        users_community = self.exp.get_community_users()
+        for user in users_community:
+            self.delete_user(user)
+
+    def delete_basic_users(self):
+        """
+        It deletes the basic users.
+        :return:
+        """
+        users_basic = self.exp.get_basic_users()
+        for user in users_basic:
+            self.delete_user(user)
+
+    def update_quota(self, user, role):
+        """ It updates the quota for the user according to role requirements.
+        the user should be registrated in keystone.
+        :param user: the user
+        :param role: the role
+        :return: nothing
+        """
+        kargs = self.get_nova_quota(user, role)
+        self.nova_c.quotas.update(user.default_project_id, **kargs)
+        self.nova_c.quotas.update(user.cloud_project_id, **kargs)
+        self.neutron_c.update_quota(user.cloud_project_id, self.get_neutron_quota(role))
+
+    def get_neutron_quota(self, role):
+        """
+        It gets the neutron quota parameters
+        :param role: the user role
+        :return: the quota
+        """
+        if role == 'community':
```

Comment:
how do you check this code? I though that in production environment role are a number sequence not a string.